vllm/v1/attention/backends/mla/flashattn_mla.py

```diff
@@ -142,10 +135,11 @@ def _build_decode(
         query_start_loc_cpu: torch.Tensor,
         query_start_loc_device: torch.Tensor,
         num_decode_tokens: int,
+        dcp_tot_seq_lens_device: Optional[torch.Tensor],
     ) -> FlashAttnMLADecodeMetadata:
         query_lens_cpu = query_start_loc_cpu[1:] - query_start_loc_cpu[:-1]
         max_query_len = query_lens_cpu.max().item()
-        max_seq_len = seq_lens_cpu.max().item()
+        max_seq_len = seq_lens_device.max().item()
```

Comment:
@minosfuture this introduces a CPU sync and will be problematic for performance, especially in the async scheduling case. It looks like it should not be too hard to ensure we obtain this from the cpu-side tensor in the DCP case too?